Extensions/ShareTo/ShareViewController.swift

```diff
@@ -367,12 +367,14 @@ extension ShareViewController {
         }
 
         gesture.isEnabled = false
-        sendToDevice = SendToDevice()
-        guard let sendToDevice = sendToDevice else { return }
-        sendToDevice.sharedItem = item
-        sendToDevice.delegate = delegate
-        let vc = sendToDevice.initialViewController()
-        navigationController?.pushViewController(vc, animated: true)
+        RustFirefoxAccounts.shared.accountManager.uponQueue(.main) { _ in
```

Comment:
This probably looks strange. Basically the app or an app extension needs to init `RustFirefoxAccounts.shared.accountManager` on startup (we do this in app did finish launching). However, app extensions need to also init this in their startup code, which this does do. However, this code was not waiting for that to complete before showing the "Share To Device" VC –which expects that init is complete. We don't put these wait calls everywhere in the app, since in most cases, the caller is expected to handle the case where this is not initialized graciously, or perhaps it is sitting among other code that could not have ran unless the init was complete.
It is a case-by-case basis. In this case, we must wait for the init to complete. The init is only supposed to take a sec to run, but perhaps it is a bit slower and the user was fast enough tapping on the 'Send To Device' button that init was not completed.
This is using Deferred syntax, when a Deferred is filled with a value, uponQueue() is called, either waiting for it to be filled or called immediately if already filled with a value.

Reply:
Thanks for this explanation